migrations/3.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-await-in-loop */
+import _ from "lodash";
+
+/**
+ * @summary Performs migration up from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function up({ db, progress }) {
+  const affectedGroups = [
+    "owner",
+    "shop manager"
+  ];
+
+  const newShopPermissions = [
+    "reaction:legacy:groups/create",
+    "reaction:legacy:groups/delete",
+    "reaction:legacy:groups/manage:accounts",
+    "reaction:legacy:groups/read",
+    "reaction:legacy:groups/update",
+    "reaction:legacy:inventory/update:settings",
+    "reaction:legacy:navigationTreeItems/update:settings",
+    "reaction:legacy:navigationTrees/read:drafts",
+    "reaction:legacy:shipping-rates/update:settings",
+    "reaction:legacy:sitemaps/update:settings",
+    "reaction:legacy:tags/read:invisible",
+    "reaction:legacy:taxes/update:settings"
+  ];
+
+  // get all groups that need new permissions
+  const groups = await db.collection("Groups").find({ slug: { $in: affectedGroups } }).toArray();
```

Comment:
I think you could just do an `updateMany` with `$addToSet` and with `$each` set to `newShopPermissions`, and then no looping in code would be necessary. It may not make much of a difference for performance since there are only two groups affected, but less code means less that might go wrong.

https://docs.mongodb.com/manual/reference/operator/update/addToSet/#each-modifier

Reply:
Is this what you are suggesting?

https://github.com/reactioncommerce/plugin-simple-authorization/pull/7/commits/30d3ded0a797b52246017db8e5ce41b23a8c8f94